ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrations.scala

```diff
@@ -68,6 +69,49 @@ private[platform] class FlywayMigrations(jdbcUrl: String)(implicit loggingContex
       }
     }
 
+  def validateAndWaitOnly(
+      // TODO append-only: remove after removing support for the current (mutating) schema
+      enableAppendOnlySchema: Boolean = false
+  )(implicit resourceContext: ResourceContext): Future[Unit] =
+    dataSource.use { ds =>
+      val flyway = configurationBase(dbType, enableAppendOnlySchema)
+        .dataSource(ds)
+        .ignoreFutureMigrations(false)
+        .load()
+      logger.info("Running Flyway validation...")
+
+      @tailrec
+      def flywayMigrationDone(
+          retries: Int,
+          needLessThan: Option[Int],
```

Comment:
Can you think of a better name? I found myself having to go to the implementation to figure out what this argument is about, which is not a good sign.

Reply:
Sure, changing `needLessThan` to be more "declarative" `pendingMigrationsSoFar`.